src/utp_wrapper.rs

```diff
@@ -47,7 +49,13 @@ impl UtpWrapper {
                                     }
                                 },
                                 Err(TryRecvError::Disconnected) => break 'outer,
-                                Err(TryRecvError::Empty) => break,
+                                Err(TryRecvError::Empty) => {
+                                    if timeout < ::time::SteadyTime::now() {
```

Comment:
> It will then require some changes to the utp_wrapper API and better to be `re-designed`. I think it shall be better to be left over to a better crust understanding guy, such as you and @vinipsmaker , and better for another PR.

Okay, I've opened a [new PR to investigate the hanging issue under Windows](https://github.com/maidsafe/crust/pull/451).